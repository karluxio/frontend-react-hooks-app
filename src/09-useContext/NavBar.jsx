import { NavLink } from 'react-router-dom'

export const NavBar = () => {
  return (
    <>
      <nav className="navbar navbar-expand-lg navbar-dark bg-dark p-3 rounded-3">
        <NavLink className="navbar-brand" to="/">UseContext</NavLink>

        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li className="nav-item">
              <NavLink className="nav-link" to="/about">About</NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link" to="/login">Login</NavLink>
            </li>
          </ul>
        </div>
      </nav>
    </>
  )
}