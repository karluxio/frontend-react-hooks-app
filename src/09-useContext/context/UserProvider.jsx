import { useState } from "react"
import { UserContext } from "./UserContext"

// const user = {
//   id: 1,
//   name: "Luciano Figueroa",
//   email: "lucianofigueroa@email.com"
// }

export const UserProvider = ({ children }) => {

  const [user, setUser] = useState()

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  )
}