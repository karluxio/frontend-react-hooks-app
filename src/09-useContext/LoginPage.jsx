import { useContext } from "react"
import { UserContext } from "./context/UserContext";

export const LoginPage = () => {

  const { user, setUser } = useContext(UserContext)

  return (
    <>
      <h1>Login Page</h1>
      <hr />

      <pre aria-label="pre">
        {JSON.stringify(user, null, 2)}
      </pre>

      <button onClick={() => setUser({ id: 123, name: 'Luciano', email: 'luc@email.com' })} className="btn btn-primary">login</button>
    </>
  )
}