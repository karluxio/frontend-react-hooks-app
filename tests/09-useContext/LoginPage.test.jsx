import { render, screen, fireEvent } from '@testing-library/react'
import { LoginPage } from '../../src/09-useContext/LoginPage'
import { UserContext } from '../../src/09-useContext/context/UserContext'

describe("LoginPage", () => {
  const user = {
    id: 1,
    name: 'Luciano'
  }

  it("should match against snapshot", () => {
    const { container } = render(
      <UserContext.Provider value={user}>
        <LoginPage />
      </UserContext.Provider>
    )

    expect(container).toMatchSnapshot()
  })

  it("should render user in pre tag", () => {
    render(
      <UserContext.Provider value={{ user }}>
        <LoginPage />
      </UserContext.Provider>
    )

    const preTag = screen.getByLabelText('pre')
    expect(preTag.innerHTML).toContain(user.name)
    expect(preTag.innerHTML).toContain(`${user.id}`)
  })

  it("should call setUser function when login button is clicked", () => {
    const setUserMock = jest.fn()
    render(
      <UserContext.Provider value={{ user, setUser: setUserMock }}>
        <LoginPage />
      </UserContext.Provider>
    )

    const button = screen.getByRole("button", { name: /login/i })
    fireEvent.click(button)
    expect(setUserMock).toHaveBeenCalledTimes(1)
    expect(setUserMock).toHaveBeenCalledWith({ id: 123, name: 'Luciano', email: 'luc@email.com' })
  })
})