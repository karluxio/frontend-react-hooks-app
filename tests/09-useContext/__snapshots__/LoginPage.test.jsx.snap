// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match against snapshot 1`] = `
<div>
  <h1>
    Login Page
  </h1>
  <hr />
  <pre
    aria-label="pre"
  />
  <button
    class="btn btn-primary"
  >
    login
  </button>
</div>
`;
